### ISSUE TITLE
CsvSource does not handle trailing spaces correctly

### ISSUE THREAD

**Issue by sormuras** [junit-team/junit5#1100]
## Overview

Copied from https://github.com/junit-team/junit5/pull/1016#issuecomment-334091550

I maybe found a bug with the CsvSource parser for empty strings. Consider the following example:

```java
@ParameterizedTest
    @CsvSource({
        "  '', Content",
        "  '',      ''",
        "  '', ''     ",
    })
void contentTest(final String one, final String two) {
  Assertions.assertNotNull(two);
}
```

Where in the second test the string two is, as expected an empty string, in the third test the string two is ```null```. Obviously the parser is confused by the trailing blanks of the second parameter.

I'm currently using junit jupiter version 5.0.0.

## Deliverables

- [ ] Upgrade to `univocity-parsers` version 2.5.8 when available and remove [TODO](https://github.com/junit-team/junit5/commit/62055132ae226f7e99031d779926ed553a8ed543#diff-a25bf3dde2d64a209206c59a9742f823R26)

**Comment by sormuras** [comment]
Good catch. Smells like a bug in https://github.com/uniVocity/univocity-parsers because if you switch the content inside the input, everything works as expected:

```java
	@ParameterizedTest
	@CsvSource(value = { 
		"'', 1",
		" '', 2",
		"'' , 3",
		" '' , 4" })
	void contentTestTwo(String one, String two) {
		System.out.printf("`%s`, `%s`\n", one, two);
		assertNotNull(one);
		assertNotNull(two);
	}
[...]
``, `1`
``, `2`
``, `3`
``, `4`
```
So, yes, please file a new bug -- even if the real issue lies potentially in an external library.

Edit: a simple fix on our side is `trim()` each line before feeding to the parser. I'm not sure whether this is breaking existing expectations. Like insert `.map(String::trim)` before line 50 here:
https://github.com/junit-team/junit5/blob/f32a8a2842eb1a4a269f5a0a808ab1f52fad0430/junit-jupiter-params/src/main/java/org/junit/jupiter/params/provider/CsvArgumentsProvider.java#L49-L53

**Comment by sormuras** [comment]
Tentatively set to 5.1 M1 as the bug was fixed in the `univocity-parsers` 2.5.8-SNAPSHOT release.
